src/main/java/commons/headers/Headers.java

```diff
@@ -0,0 +1,85 @@
+package commons.headers;
+
+import commons.header.Header;
+
+import java.util.*;
+
+public class Headers implements HeadersInterface {
+    private final HashMap<String, ArrayList<Header>> hashMap = new LinkedHashMap<>();
```

Comment:
hmm not 100% certain if we should have the values be `ArrayList<Header>`
I was thinking each key-value pair would be at most 1 key, 1 value, lmk your thought process on the decision though

Reply:
My thought process was.... For each `key` there will be a list of Header that all share the same key with different values... Take the following example:

```javascript
Key: "Accept"
Value: [
    Header(key="Accept", value="text/html"),
    Header(key="Accept", value="application/json"),
    Header(key="Accept", value="image/png")
]

Key: "Set-Cookie"
Value: [
    Header(key="Set-Cookie", value="sessionId=abc123"),
    Header(key="Set-Cookie", value="userId=kyoshi"),
    Header(key="Set-Cookie", value="theme=dark")
]

Key: "Authorization"
Value: [
    Header(key="Authorization", value="Bearer token123")
]

Key: "Content-Type"
Value: [
    Header(key="Content-Type", value="application/json")
]
```

Reply:
ok yeah I get your justification, lgtm then